code/pages/ProductPage.php

```diff
@@ -407,15 +407,15 @@ public function PurchaseForm() {
 
 			$optionsSet = CompositeField::create();
 
-			foreach ($groupedBy as $id => $set) {
+			foreach($groupedBy as $id => $set){
 				$group = OptionGroup::get()->byID($id);
-				$title = $group->Title;
+				$title = preg_replace('/[^a-zA-Z0-9\:\-\_\.]/','-', $group->Title);
```

Comment:
The `pre_replace()` may be better suited in the `ProductPage::getGeneratedValue()` function as that would handle any mishaps for custom fields added via Extensions as well. Something to the effect of:

```php
public static function getGeneratedValue($productCode = null, $optionName = null, $optionValue = null, $method = 'name', $output = false, $urlEncode = false){
$optionName = ($optionName != null) ? preg_replace('/[^a-zA-Z0-9\:\-\_\.]/','-', $optionName) : $optionValue;
		return (SiteConfig::current_site_config()->CartValidation)
			? FoxyCart_Helper::fc_hash_value($productCode, $optionName, $optionValue, $method, $output, $urlEncode):
			$optionValue;
	}
```